packages/manager/modules/server-sidebar/src/controller.js

```diff
@@ -343,88 +362,119 @@ export default class OvhManagerServerSidebarController {
+                if (
+                  has(typeServices.type, 'loadOnStateParams') &&
+                  has(typeServices.type, 'loadOnState')
+                ) {
+                  loadOnStateParams = zipObject(
+                    get(typeServices.type, 'loadOnStateParams', []),
+                    get(service, 'stateParams', []),
+                  );
+                }
+
+                let link = null;
+                let state = null;
+                if (isExternal) {
+                  link = service.url;
+                } else {
+                  state = get(typeServices.type, 'state', '');
+                  if (
+                    has(typeServices.type, 'getState') &&
+                    isFunction(typeServices.type.getState)
+                  ) {
+                    state = typeServices.type.getState(service.extraParams);
+                  }
+                }
+
+                const prefixTranslationPart = get(typeServices.type, 'prefix');
+                let prefix;
+                if (prefixTranslationPart) {
+                  prefix = this.$translate.instant(
+                    `server_sidebar_item_${prefixTranslationPart}_prefix`,
+                  );
+                }
+
+                if (
+                  has(typeServices.type, 'getPrefix') &&
+                  isFunction(typeServices.type.getPrefix)
+                ) {
+                  prefix = this.$translate.instant(
+                    typeServices.type.getPrefix(service.extraParams),
+                  );
                 }
-              }
-
-              const menuItem = this.SidebarMenu.addMenuItem(
-                {
-                  id: service.serviceName,
-                  title: service.displayName,
-                  allowSubItems: hasSubItems && !isExternal,
-                  infiniteScroll: hasSubItems && !isExternal,
-                  allowSearch: false,
-                  state,
-                  stateParams,
-                  url: link,
-                  target: isExternal ? '_self' : null,
-                  icon: get(typeServices.type, 'icon'),
-                  loadOnState: get(typeServices.type, 'loadOnState'),
-                  loadOnStateParams,
-                  namespace: typeServices.type.namespace,
-                },
-                parent,
-              );
 
-              // add serviceName in item searchKeys
-              if (has(service, 'serviceName')) {
-                menuItem.addSearchKey(service.serviceName);
-              }
-
-              // add searchKeys from type definition in item searchKeys
-              if (
-                has(typeServices.type, 'searchKeys') &&
-                isArray(typeServices.type.searchKeys)
-              ) {
-                each(typeServices.type.searchKeys, (key) =>
-                  menuItem.addSearchKey(key),
+                let icon = get(typeServices.type, 'icon');
+                if (
+                  has(typeServices.type, 'getIcon') &&
```

Comment:
you can use es6 "?" syntax instead of has

Reply:
Don't want to refactor the code of this module that will be removed soon.